src/sentry/db/models/base.py

```diff
@@ -92,6 +94,55 @@ class Meta:
     __repr__ = sane_repr("id")
 
 
+@dataclass
+class SnowflakeBitSegment:
+    length: int
+    name: str
+
+    def validate(self, value):
+        if value >> self.length != 0:
+            raise Exception(f"{self.name} exceed max bit value of {self.length}")
+        return True
+
+
+class Snowflake:
+    SENTRY_EPOCH_START = datetime(2022, 4, 26, 0, 0).timestamp()
+    SNOWFLAKE_ID_LENGTH = SnowflakeBitSegment(53, "Snowflake ID")
+
+    SEGMENT_LENGTH = {
+        "VERSION_ID": SnowflakeBitSegment(5, "Version ID"),
+        "TIME_DIFFERENCE": SnowflakeBitSegment(32, "Time difference"),
+        "REGION_ID": SnowflakeBitSegment(12, "Region ID"),
+        "REGION_SEQUENCE": SnowflakeBitSegment(4, "Region sequence"),
```

Comment:
Should this have a way to enumerate the local sequence?

Reply:
yes, i plan to do that in a followup pr as I am implementing this https://www.notion.so/sentry/Assigning-region-sequence-values-at-random-205a213bf1334ba6975340389bdd0b4b